include/libtorrent/kademlia/node.hpp

```diff
@@ -234,6 +234,10 @@ class TORRENT_EXTRA_EXPORT node : boost::noncopyable
 	libtorrent::dht_settings const& m_settings;
 
 private:
+
+	void update_counters();
```

Comment:
if this is the only contact surface of the counters&, maybe it could be passed in here instead of relying on a member (if the member could be removed as a result). could it?